main/plugins/config.ts

```diff
@@ -0,0 +1,84 @@
+import {ValidateFunction} from 'ajv';
+import Store, {Schema as JSONSchema} from 'electron-store';
+import Ajv, {Schema} from '../utils/ajv';
+import {Service} from './service';
+
+export default class PluginConfig extends Store {
+  servicesWithNoConfig: Service[];
+  validators: {
+    title: string;
+    description?: string;
+    config: {[key: string]: Schema}
+    validate: ValidateFunction
+  }[]
+
+  constructor(name: string, services: Service[]) {
+    const defaults = {};
+
+    const validators = services
```

Comment:
One thing I noticed here, but hasn't come up as an issue yet, is that we allow each service exported by a plugin to have its own `config` schema object, but we actually save all of them (and validate them) against the same object (electron-store instance). So the plugins need to make sure their different services don't have conflicting field schemas